File: hdlibrary/src/main/java/com/hd/base/BaseActivity.java
package com.hd.base;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.app.AppCompatDelegate;
import android.util.Log;

/**
 * Created by hanyw on 2017/12/4/004.
 * 基础activity
 */

public abstract class BaseActivity extends AppCompatActivity {
    //
    protected Context mContext;

    /** 日志输出标志 **/
    protected final String TAG = this.getClass().getSimpleName();
    /** ViewModel标志 **/
    protected final String VIEWMODEL_TAG = this.getClass().getSimpleName();

    protected BaseViewModel mViewModel;


    protected Fragment mFragment;


    public abstract Fragment findOrCreateViewFragment();
    public abstract BaseViewModel findOrCreateViewModel();

    //矢量支持
    static {
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
    }

    /**
     * [页面跳转]
     *
     * @param clz
     */
    public void startActivity(Class<?> clz) {
        startActivity(new Intent(BaseActivity.this,clz));
    }


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
        mContext=this;

    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(TAG, "onRestart()");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(TAG, "onStart()");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG, "onResume()");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "onPause()");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "onStop()");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy()");
    }
}

File: app/src/main/java/com/hd/share/BaseActivity.kt
package com.hd.share

import android.content.Context
import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.hd.util.DeviceUtil

/**
 * Created by whs on 2017/5/17
 * 基础Activity
 */

abstract class BaseActivity : AppCompatActivity() {

    protected var mContext: Context?=null

    /** 日志输出标志  */
    protected val TAG = this.javaClass.simpleName
    /** ViewModel标志  */
    protected val VIEWMODEL_TAG = this.javaClass.simpleName

    protected var mViewModel: BaseViewModel? = null


    protected var mFragment: Fragment? = null


    abstract fun findOrCreateViewFragment(): Fragment
    abstract fun findOrCreateViewModel(): BaseViewModel

    /**
     * [页面跳转]
     * @param clz
     */
    fun startActivity(clz: Class<*>) {
        startActivity(Intent(this@BaseActivity, clz))
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = this
        //如果是平板使用横屏模式
        if (DeviceUtil.isTablet(mContext as BaseActivity)) {
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        } else {
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        }

    }

    override fun onRestart() {
        super.onRestart()
        Log.d(TAG, "onRestart()")
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "onStart()")
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume()")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause()")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "onStop()")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "onDestroy()")
    }

}
